subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -1273,33 +1273,24 @@ static int prop_set(struct net_buf_simple *buf,
 	LOG_DBG("Set Prop: 0x%04x: %s", id, bt_mesh_sensor_ch_str(&val));
 
 #if CONFIG_BT_MESH_LIGHT_CTRL_SRV_REG
-	/* Regulator coefficients are raw IEEE-754 floats, pull them straight
-	 * from the buffer instead of using sensor to decode them:
+	/* Regulator coefficients are raw IEEE-754 floats, status will always
+	 * be BT_MESH_SENSOR_VALUE_NUMBER:
 	 */
 	if (srv->reg) {
 		switch (id) {
 		case BT_MESH_LIGHT_CTRL_COEFF_KID:
-			status = bt_mesh_sensor_value_to_float(
-				&val, &srv->reg->cfg.ki.down);
-			break;
+			(void)bt_mesh_sensor_value_to_float(&val, &srv->reg->cfg.ki.down);
```

Comment:
Why did you change it to ignore error code? What if a client (not PTS but another 3rd-party one) sent wrong data?

Reply:
This was my suggestion, and is explained in the comment above: `val->format` will always be `&bt_mesh_sensor_format_coefficient` here (because it is looked up from the ID above). `bt_mesh_sensor_format_coefficient.cb->to_float` always returns BT_MESH_SENSOR_VALUE_NUMBER, no matter what `val->raw` is (because it simply reinterprets the bytes as a float).

Reply:
Why do we need return code there then? :/

Reply:
Because other formats might have special `val->raw` values for which they return things like "Value is invalid", "Value is not know", etc. But since we know the format here we know that we can ignore it, it will always be a number _in this case with this format_. But `bt_mesh_sensor_value_to_float` needs to be general for all formats.